File: 1189. Maximum Number of Balloons/Solution.cs
public class Solution {
    public int MaxNumberOfBalloons(string text) {

        var char_frequencies = new int[26];
        var expected_frequencies = new int[26];

        foreach(var c in text) char_frequencies[c-'a']++;
        foreach(var c in "balloon") expected_frequencies[c-'a']++;

        int max_balloons = char_frequencies['b'-'a'];

        for (int c = 0; c < 26; ++c) {
            if (expected_frequencies[c] == 0) continue;
            max_balloons = Math.Min(max_balloons, char_frequencies[c]/expected_frequencies[c]);
        }
            

        return max_balloons;
    }
}

File: 1189. Maximum Number of Balloons/solution.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:

        char_frequencies = collections.Counter(text)
        expected_frequencies = collections.Counter('balloon')

        max_balloons = char_frequencies['b']

        for char in expected_frequencies.elements():
            max_balloons = min(max_balloons, char_frequencies[char]//expected_frequencies[char])

        return max_balloons
